src/main/java/cope/cosmos/client/ui/altmanager/AltManagerGUI.java

```diff
@@ -0,0 +1,263 @@
+
+                for (AltEntry altEntry : AltManager.getAltEntries()) {
+                    altEntry.setOffset(altEntry.getOffset() + 16);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void actionPerformed(GuiButton button) throws IOException {
+        super.actionPerformed(button);
+
+        switch (button.id) {
+            case 0:
+                // Display add alt gui
+                mc.displayGuiScreen(new AddAltGUI(this));
+                break;
+            case 1:
+                // If there is a selected entry, remove it
+                if(getSelectedAltEntry() != null) {
+                    // Remove the entry
+                    AltManager.getAltEntries().remove(getSelectedAltEntry());
+                    // Decrease the offset
+                    altEntryOffset -= 32;
+                    // Refresh positions of all other alts
+                    if(!AltManager.getAltEntries().isEmpty())
+                        getFirstAlt().setOffset(41);
+                }
+                break;
+            case 2:
+                // Display multiplayer gui
+                mc.displayGuiScreen(lastScreen);
+                break;
+        }
+    }
+
+    /**
+     * Called when the GUI is closed.
+     */
+    @Override
+    public void onGuiClosed() {
+        // Save the alts to alts.toml
+        AltManager.saveAlts();
+    }
+
+    /**
+     * Refreshes the alt entries' offsets
+     */
+    private void refreshOffsets() {
+        // Get the first alt's offset
+        float altOffset = getFirstAlt().getOffset();
+
+        for(AltEntry altEntry : AltManager.getAltEntries()) {
+            // Set the offset to a new offset
+            altEntry.setOffset(altOffset);
+            // Increase offset for next entry
+            altOffset += 32;
+        }
+    }
+
+    /**
+     * Gets the first alt
+     * @return The first alt
+     */
+    private AltEntry getFirstAlt() {
+        return AltManager.getAltEntries().get(0);
+    }
+
+    /**
+     * There has got to be a better way of doing this...
+     * Gets the last alt in the list
+     * @return The last alt
+     */
+    private AltEntry getLastAlt() {
+        return AltManager.getAltEntries().get(AltManager.getAltEntries().size() - 1);
+    }
+
+    /**
+     * Gets the selected alt
+     * @return The selected alt
+     */
+    private AltEntry getSelectedAltEntry() {
+        for(AltEntry altEntry : AltManager.getAltEntries()) {
+            // If the alt is selected, return the entry
+            if(altEntry.isSelected)
+                return altEntry;
+        }
+
+        return null;
+    }
+
+    /**
+     * Sets the selected alt
+     * @param newSelected The new selected alt
+     */
+    private void setSelectedAltEntry(AltEntry newSelected) {
+        // Sets whether the alt is selected to whether it is the newSelected parameter
+        for(AltEntry altEntry : AltManager.getAltEntries())
```

Comment:
could use anyMatch(), but im sure what linus would like to do to. but def a space after for